internal/pd/get-shifts.go

```diff
@@ -22,96 +22,164 @@ package pd
-func GetNextShift(shifts []Shift, shiftPos int) (Shift, int, error) {
+// ShiftTime stores the amount of minutes that passed since midnight or the length of a time period in minutes
+//  3:30 am would be stored as 210 (3 * 60 + 30) and a time period of 6:15 h would be stored as 375 (6 * 60 + 15)
+type ShiftTime int
+
+// GetProbablyOwnShift returns the shift the user probably belongs to because of their time zone
+func GetProbablyOwnShift() (Shift, error) {
+
+	timeOffsetString := time.Now().String()
+	if offsetLocation := strings.Index(timeOffsetString, " +"); offsetLocation != -1 {
+		timeOffsetString = timeOffsetString[offsetLocation+1 : offsetLocation+6]
+	} else {
+		offsetLocation := strings.Index(timeOffsetString, " -")
+		timeOffsetString = timeOffsetString[offsetLocation+1 : offsetLocation+6]
+	}
+
+	timeOffset, err := strconv.Atoi(timeOffsetString[1:3])
+	if err != nil {
+		return Shift{}, err
+	}
+	timeOffset *= 60
+
+	minutes, err := strconv.Atoi(timeOffsetString[3:])
+	if err != nil {
+		return Shift{}, err
+	}
+	timeOffset += minutes
+
+	var middayInUTC ShiftTime
+	if timeOffsetString[0] == '+' {
+		middayInUTC = ShiftTime((36*60 - timeOffset) % (24 * 60))
+	} else {
+		middayInUTC = ShiftTime((12*60 + timeOffset) % (24 * 60))
+	}
+
+	return GetShiftByTime(middayInUTC)
+}
+
+// GetShiftByTime returns the shift which is active at a specific time
+func GetShiftByTime(time ShiftTime) (Shift, error) {
+
+	shifts, _, err := LoadShifts()
+	if err != nil {
+		return Shift{}, err
+	}
+
+	rightShift := Shift{}
+	for i, shift := range shifts {
+		if shift.Start < shift.End { // shift starts and ends during the same day
+			if time >= shift.Start && time < shift.End {
+				rightShift = shifts[i]
+			}
+		} else {
+			if time >= shift.Start || time < shift.End {
+				rightShift = shifts[i]
+			}
+		}
+	}
+
+	return rightShift, nil
+}
+
+// GetTimeUntilShift returns information about the next shift and time until it starts
+func GetTimeUntilShift(shifts []Shift, shiftPos int) (ShiftTime, error) {
 
 	timeInUTC := time.Now().UTC()
-	currentTime := timeInUTC.Hour()*60 + timeInUTC.Minute()
+	currentTime := ShiftTime(timeInUTC.Hour()*60 + timeInUTC.Minute())
 
-	nextShift := shifts[(shiftPos+1)%len(shifts)]
+	nextShift := shifts[shiftPos%len(shifts)]
 
 	timeUntilNextShift := nextShift.Start - currentTime
 	if currentTime > nextShift.Start { // prevents timeUntilNextShift from being negative if the next shift starts during the next day
 		timeUntilNextShift += 24 * 60
 	}
 
-	return nextShift, timeUntilNextShift, nil
+	return timeUntilNextShift, nil
 }
 
-// GetCurrentShift returns all shifts in a slice and the position of the current shift
-func GetCurrentShift() ([]Shift, int, error) {
+// GetCurrentAndOwnShift returns all shifts in a slice and the position of the current shift
+func GetCurrentAndOwnShift() ([]Shift, int, int, error) { // first int is the position of current-shift in []Shift and second int is the position of own-shift
```

Comment:
```suggestion
// GetCurrentAndOwnShift returns all shifts in a slice, the position of the current shift, and
// the position of your own-shift, or an error otherwise
func GetCurrentAndOwnShift() ([]Shift, int, int, error) {
```